api/bazel/envoy_http_archive.bzl

```diff
@@ -0,0 +1,32 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+def envoy_http_archive(name, locations, **kwargs):
+    # `existing_rule_keys` contains the names of repositories that have already
+    # been defined in the Bazel workspace. By skipping repos with existing keys,
+    # users can override dependency versions by using standard Bazel repository
+    # rules in their WORKSPACE files.
+    existing_rule_keys = native.existing_rules().keys()
+    if name in existing_rule_keys:
+        # This repository has already been defined, probably because the user
+        # wants to override the version. Do nothing.
+        return
+
+    loc_key = kwargs.pop("repository_key", name)
+    location = locations[loc_key]
+
+    # Git tags are mutable. We want to depend on commit IDs instead. Give the
```

Comment:
This is the leftover from when we were using `git_repository` which is no longer needed here. For HTTP archive as long as URL and SHA256 is verified it is fine.

Reply:
Removed the tag check